ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -415,6 +411,27 @@ public partial class PrincipiaPluginAdapter
     return active_vessel != null && is_manageable(active_vessel);
   }
 
+  private bool JustAboveTheGround(Vessel vessel,
+                                  out double height,
+                                  out double vertical_speed) {
+    height = vessel.altitude - vessel.terrainAltitude;
+    vertical_speed = vessel.verticalSpeed;
+    double Δt = Planetarium.TimeScale * Planetarium.fetch.fixedDeltaTime;
+    return height + vertical_speed * Δt < 0;
+  }
+
+  private bool TouchesTheGround(Vessel vessel) {
+    return vessel.parts
+        .Where(part =>
+               part.Modules.OfType<ModuleWheelBase>()
+                   .Where(wheel => wheel.isGrounded)
+                   .Any() ||
+               part.currentCollisions
+                   .Where(collider => collider.gameObject.layer == 15)
```

Comment:
Could we have a named constant for `15`?

Reply:
Added an enum.